linera-views/src/simp_store.rs

```diff
@@ -0,0 +1,333 @@
+    async fn contains_key(&self, key: &[u8]) -> Result<bool, Self::Error>;
+
+    /// Retrieves multiple `Vec<u8>` from the database using the provided `keys`.
+    async fn read_multi_values_bytes(
+        &self,
+        keys: Vec<Vec<u8>>,
+    ) -> Result<Vec<Option<Vec<u8>>>, Self::Error>;
+
+    /// Finds the `key` matching the prefix. The prefix is not included in the returned keys.
+    async fn find_keys_by_prefix(&self, key_prefix: &[u8]) -> Result<Self::Keys, Self::Error>;
+
+    /// Finds the `(key,value)` pairs matching the prefix. The prefix is not included in the returned keys.
+    async fn find_key_values_by_prefix(
+        &self,
+        key_prefix: &[u8],
+    ) -> Result<Self::KeyValues, Self::Error>;
+
+    /// Writes the simplified `batch` in the database. After the writing, the
+    /// simplified batch must be empty.
+    async fn write_simplified_batch(
+        &self,
+        simp_batch: &mut Self::SimpBatch,
+    ) -> Result<(), Self::Error>;
+
+    /// Reads a single `key` and deserializes the result if present.
+    async fn read_value<V: DeserializeOwned>(&self, key: &[u8]) -> Result<Option<V>, Self::Error>
+    where
+        Self::Error: From<bcs::Error>,
+    {
+        from_bytes_opt(&self.read_value_bytes(key).await?)
+    }
+
+    /// Reads multiple `keys` and deserializes the results if present.
+    async fn read_multi_values<V: DeserializeOwned + Send>(
+        &self,
+        keys: Vec<Vec<u8>>,
+    ) -> Result<Vec<Option<V>>, Self::Error>
+    where
+        Self::Error: From<bcs::Error>,
+    {
+        let mut values = Vec::with_capacity(keys.len());
+        for entry in self.read_multi_values_bytes(keys).await? {
+            values.push(from_bytes_opt(&entry)?);
+        }
+        Ok(values)
+    }
+}
+
+/// The header that contains the current state of the journal.
+#[derive(Serialize, Deserialize)]
+struct JournalHeader {
+    block_count: u32,
+}
+
+/// A KeyValueStore built from a SimplifiedKeyValueStore by using journaling
+/// for bypassing the limits on the batch size
+#[derive(Clone)]
+pub struct StoreFromSimplifiedStore<K> {
+    /// The inner client that is called by the LRU cache one
+    pub client: K,
+}
+
+#[async_trait]
+impl<K> DeletePrefixExpander for &StoreFromSimplifiedStore<K>
+where
+    K: SimplifiedKeyValueStore + Send + Sync,
+{
+    type Error = K::Error;
+    async fn expand_delete_prefix(&self, key_prefix: &[u8]) -> Result<Vec<Vec<u8>>, Self::Error> {
+        let mut vector_list = Vec::new();
+        for key in self
+            .client
+            .find_keys_by_prefix(key_prefix)
+            .await?
+            .iterator()
+        {
+            vector_list.push(key?.to_vec());
+        }
+        Ok(vector_list)
+    }
+}
+
+#[async_trait]
+impl<K> KeyValueStore for StoreFromSimplifiedStore<K>
+where
+    K: SimplifiedKeyValueStore + Send + Sync,
+    K::Error: From<JournalConsistencyError>,
+{
+    /// The reqding constants do not change
```

Comment:
Did you mean "reading"? I don't really understand the comment. Should it be "the size constants"?

Reply:
Yes, thanks.